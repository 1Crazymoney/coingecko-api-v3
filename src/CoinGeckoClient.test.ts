import { CoinGeckoClient } from './CoinGeckoClient';

const client = new CoinGeckoClient();
jest.setTimeout(30000)
describe('CoinGeckoClient test', () => {
  it('ping should successful', async () => {
    const ping = await client.ping();
    expect(ping).toEqual({ gecko_says: '(V3) To the Moon!' })
  })

  it('/search/trending should successful', async () => {
    const trending = await client.trending();
    expect(trending.coins?.length).toBeGreaterThan(1);
    trending.coins?.forEach(({ item }) => expect(item).toMatchObject({
      id: expect.any(String),
      large: expect.any(String),
      market_cap_rank: expect.any(Number),
      name: expect.any(String),
      score: expect.any(Number),
      symbol: expect.any(String),
      thumb: expect.any(String),
    }))
  })

  it('/coins/list should successful', async () => {
    const list = await client.coinList({ include_platform: true });
    expect(list.length).toBeGreaterThan(1)
  })

  it('/coins/market should successful', async () => {
    const list = await client.coinMarket({ vs_currency: 'usd', ids: 'origin-protocol,bitcorn' });
    expect(list.length).toEqual(2)
    expect(list).toMatchSnapshot();
  })

  it.only('/coins/{id}/tickers should successful', async () => {
    const ticker = await client.coinTickers({ id: 'origin-protocol' });
    expect(ticker.name).toEqual("Origin Protocol")
    expect(ticker.tickers.length).toBeGreaterThan(0)
  })


  it.only('/simple/market should successful', async () => {
    const price = await client.simpleTokenPrice({ contract_addresses: '0x8207c1ffc5b6804f6024322ccf34f29c3541ae26', id: 'ethereum', vs_currencies: 'btc,eth' });
    expect(price).toMatchObject({
      '0x8207c1ffc5b6804f6024322ccf34f29c3541ae26': {
        btc: expect.any(Number),
        eth: expect.any(Number)
      }
    });
  })

  it('/simple/supported_vs_currencies should successful', async () => {
    const list = await client.simpleSupportedCurrencies();
    expect(list).toMatchSnapshot();
  })


})
