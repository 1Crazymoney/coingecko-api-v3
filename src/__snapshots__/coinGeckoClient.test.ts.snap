// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinGeckoClient test /coins/{id}/history should successful 1`] = `
Object {
  "community_data": Object {
    "facebook_likes": null,
    "reddit_accounts_active_48h": "9113.07692307692",
    "reddit_average_comments_48h": 1072.333,
    "reddit_average_posts_48h": 6.917,
    "reddit_subscribers": 2648989,
    "twitter_followers": 83714,
  },
  "developer_data": Object {
    "closed_issues": 5543,
    "code_additions_deletions_4_weeks": Object {
      "additions": 1942,
      "deletions": -1331,
    },
    "commit_count_4_weeks": 332,
    "forks": 28208,
    "pull_request_contributors": 720,
    "pull_requests_merged": 8450,
    "stars": 51244,
    "subscribers": 3752,
    "total_issues": 6109,
  },
  "id": "bitcoin",
  "image": Object {
    "small": "https://assets.coingecko.com/coins/images/1/small/bitcoin.png?1547033579",
    "thumb": "https://assets.coingecko.com/coins/images/1/thumb/bitcoin.png?1547033579",
  },
  "localization": Object {
    "ar": "بيتكوين",
    "de": "Bitcoin",
    "en": "Bitcoin",
    "es": "Bitcoin",
    "fr": "Bitcoin",
    "hu": "Bitcoin",
    "id": "Bitcoin",
    "it": "Bitcoin",
    "ja": "ビットコイン",
    "ko": "비트코인",
    "nl": "Bitcoin",
    "pl": "Bitcoin",
    "pt": "Bitcoin",
    "ro": "Bitcoin",
    "ru": "биткоин",
    "sv": "Bitcoin",
    "th": "บิตคอยน์",
    "tr": "Bitcoin",
    "vi": "Bitcoin",
    "zh": "比特币",
    "zh-tw": "比特幣",
  },
  "market_data": Object {
    "current_price": Object {
      "aed": 216047.9777669398,
      "ars": 5410316.137126201,
      "aud": 77443.71866877718,
      "bch": 108.88867356866139,
      "bdt": 4986355.506661704,
      "bhd": 22175.855659776604,
      "bits": 1000948.6777615082,
      "bmd": 58817.3738884188,
      "bnb": 194.4890148850541,
      "brl": 331306.5036386855,
      "btc": 1,
      "cad": 73902.559856451,
      "chf": 55518.5426565129,
      "clp": 42377917.88660569,
      "cny": 385412.60587864177,
      "czk": 1309551.1844134755,
      "dkk": 373053.9580945811,
      "dot": 1594.587856814039,
      "eos": 12269.41966053906,
      "eth": 30.733975464038206,
      "eur": 50151.45728919469,
      "gbp": 42667.71108775401,
      "hkd": 457287.6132224112,
      "huf": 18166929.56627026,
      "idr": 856839739.3317074,
      "ils": 196144.17844309928,
      "inr": 4305651.922062717,
      "jpy": 6514308.696599343,
      "krw": 66357761.22091409,
      "kwd": 17784.903429510647,
      "link": 2020.3781971342075,
      "lkr": 11731566.76407089,
      "ltc": 299.26808281595885,
      "mmk": 82914921.02210093,
      "mxn": 1202021.8496444104,
      "myr": 243886.24082832862,
      "ngn": 23259739.637827985,
      "nok": 502903.54517632176,
      "nzd": 84241.41902118336,
      "php": 2854453.6793477093,
      "pkr": 8977904.948130723,
      "pln": 232283.69038560346,
      "rub": 4453392.754385964,
      "sar": 220610.91199845573,
      "sats": 100094867.77615081,
      "sek": 513631.3636345784,
      "sgd": 79113.19100922598,
      "thb": 1837631.5064827378,
      "try": 485694.22856768314,
      "twd": 1669789.954267881,
      "uah": 1637263.2995164648,
      "usd": 58817.3738884188,
      "vef": 5889.383647447377,
      "vnd": 1358038191.0131512,
      "xag": 2404.7736228987765,
      "xau": 34.43227884801922,
      "xdr": 41476.48281439182,
      "xlm": 144402.42073839266,
      "xrp": 103349.20237563681,
      "yfi": 1.6291747927799918,
      "zar": 869847.1427497568,
    },
    "market_cap": Object {
      "aed": 4033641238570.1426,
      "ars": 101011240697450.36,
      "aud": 1445883365904.8728,
      "bch": 2032964384.3903894,
      "bdt": 93095845699323.8,
      "bhd": 414025841918.99695,
      "bits": 18687829925751.332,
      "bmd": 1098127310946.8978,
      "bnb": 3631132857.6077166,
      "brl": 6185531517101.6875,
      "btc": 18670118,
      "cad": 1379769513022.0032,
      "chf": 1036537742585.1293,
      "clp": 791200727537238.9,
      "cny": 7195698830441.736,
      "czk": 24449475140039.34,
      "dkk": 6964961417992.885,
      "dot": 29771143448.085213,
      "eos": 229071512853.78418,
      "eth": 573806948.5226982,
      "eur": 936333625461.225,
      "gbp": 796611200798.2758,
      "hkd": 8537613698800.777,
      "huf": 339178718699954.56,
      "idr": 15997299040412218,
      "ils": 3662034956545.72,
      "inr": 80387029451837.73,
      "jpy": 121622912053935.94,
      "krw": 1238907232210290.2,
      "kwd": 332046245647.56854,
      "link": 37720699345.122925,
      "lkr": 219029735810081.7,
      "ltc": 5587370419.807723,
      "mmk": 1548031359443305,
      "mxn": 22441889771439.395,
      "myr": 4553384894841.3125,
      "ngn": 434262083687525.6,
      "nok": 9389268531060.26,
      "nzd": 1572797233612.9377,
      "php": 53292987018955.89,
      "pkr": 167618544774384.47,
      "pln": 4336763908974.6826,
      "rub": 83145368224730.97,
      "sar": 4118831759098.7837,
      "sats": 1868782992575133,
      "sek": 9589558167558.488,
      "sgd": 1477052611498.788,
      "thb": 34308797066550.484,
      "try": 9067968559277.615,
      "twd": 31175175481395.94,
      "uah": 30567898999041.746,
      "usd": 1098127310946.8978,
      "vef": 109955487645.11293,
      "vnd": 25354733274722068,
      "xag": 44897407302.80765,
      "xau": 642854709.1014229,
      "xdr": 774370828369.6674,
      "xlm": 2696010234671.439,
      "xrp": 1929541803559.0193,
      "yfi": 30416885.623827998,
      "zar": 16240148797100.805,
    },
    "total_volume": Object {
      "aed": 211193735936.2901,
      "ars": 5288755254301.331,
      "aud": 75703686003.03023,
      "bch": 106442124.61419189,
      "bdt": 4874320320157.72,
      "bhd": 21677600747.66509,
      "bits": 978459011382.104,
      "bmd": 57495844477.9185,
      "bnb": 190119167.4029577,
      "brl": 323862592775.2194,
      "btc": 978459.0113821039,
      "cad": 72242091190.39264,
      "chf": 54271132544.52992,
      "clp": 41425755946340.23,
      "cny": 376753020110.4565,
      "czk": 1280127728547.5095,
      "dkk": 364672050764.6007,
      "dot": 1558760096.9007165,
      "eos": 11993746031.146273,
      "eth": 30043433.710943747,
      "eur": 49024636735.91986,
      "gbp": 41709037972.08301,
      "hkd": 447013114550.0063,
      "huf": 17758748613379.387,
      "idr": 837587963185421.1,
      "ils": 191737142164.9629,
      "inr": 4208911022729.512,
      "jpy": 6367942921276.467,
      "krw": 64866811739987.65,
      "kwd": 17385305974.01062,
      "link": 1974983630.3364513,
      "lkr": 11467977802425.322,
      "ltc": 292544022.44196784,
      "mmk": 81051959460643.12,
      "mxn": 1175014400631.427,
      "myr": 238406519127.68912,
      "ngn": 22737131639887.062,
      "nok": 491604131726.5445,
      "nzd": 82348653236.87666,
      "php": 2790318811726.366,
      "pkr": 8776186227125.959,
      "pln": 227064658862.59692,
      "rub": 4353332362152.287,
      "sar": 215654148559.19824,
      "sats": 97845901138210.39,
      "sek": 502090913792.5612,
      "sgd": 77335648052.6914,
      "thb": 1796342956502.83,
      "try": 474781480086.5948,
      "twd": 1632272527221.4238,
      "uah": 1600476692770.838,
      "usd": 57495844477.9185,
      "vef": 5757058907.573982,
      "vnd": 1327525312056481.5,
      "xag": 2350742324.6928253,
      "xau": 33658642.31581824,
      "xdr": 40544574633.87168,
      "xlm": 141157936441.70767,
      "xrp": 101027116205.14148,
      "yfi": 1592569.9214440342,
      "zar": 850303111695.8679,
    },
  },
  "name": "Bitcoin",
  "public_interest_stats": Object {
    "alexa_rank": 5018,
    "bing_matches": null,
  },
  "symbol": "btc",
}
`;

exports[`CoinGeckoClient test /coins/market should successful 1`] = `
Array [
  Object {
    "ath": 0.03418169,
    "ath_change_percentage": -88.12021,
    "ath_date": "2018-10-10T17:27:38.034Z",
    "atl": 0.00070641,
    "atl_change_percentage": 474.8347,
    "atl_date": "2020-03-16T10:22:30.944Z",
    "circulating_supply": 10646360.834599,
    "current_price": 0.00406071,
    "fully_diluted_valuation": null,
    "high_24h": 0.00422487,
    "id": "01coin",
    "image": "https://assets.coingecko.com/coins/images/5720/large/F1nTlw9I_400x400.jpg?1547041588",
    "last_updated": "2021-04-17T11:12:57.136Z",
    "low_24h": 0.00346429,
    "market_cap": 43232,
    "market_cap_change_24h": 1238.91,
    "market_cap_change_percentage_24h": 2.95027,
    "market_cap_rank": 2421,
    "max_supply": null,
    "name": "01coin",
    "price_change_24h": 0.00012003,
    "price_change_percentage_24h": 3.046,
    "roi": null,
    "symbol": "zoc",
    "total_supply": 65658824,
    "total_volume": 18509.3,
  },
  Object {
    "ath": 0.00012728,
    "ath_change_percentage": -2.1295,
    "ath_date": "2021-04-14T09:27:42.881Z",
    "atl": 0.00000177,
    "atl_change_percentage": 6930.8018,
    "atl_date": "2020-10-23T19:10:35.677Z",
    "circulating_supply": 0,
    "current_price": 0.00012457,
    "fully_diluted_valuation": null,
    "high_24h": 0.0001247,
    "id": "bitcorn",
    "image": "https://assets.coingecko.com/coins/images/10678/large/BITCORN-logo.png?1581803042",
    "last_updated": "2021-04-17T08:09:40.618Z",
    "low_24h": 0.00006068,
    "market_cap": 0,
    "market_cap_change_24h": 0,
    "market_cap_change_percentage_24h": 0,
    "market_cap_rank": null,
    "max_supply": null,
    "name": "BITCORN",
    "price_change_24h": 0.00006386,
    "price_change_percentage_24h": 105.17265,
    "roi": null,
    "symbol": "corn",
    "total_supply": 100000000000,
    "total_volume": 1155.46,
  },
]
`;

exports[`CoinGeckoClient test /simple/supported_vs_currencies should successful 1`] = `
Array [
  "btc",
  "eth",
  "ltc",
  "bch",
  "bnb",
  "eos",
  "xrp",
  "xlm",
  "link",
  "dot",
  "yfi",
  "usd",
  "aed",
  "ars",
  "aud",
  "bdt",
  "bhd",
  "bmd",
  "brl",
  "cad",
  "chf",
  "clp",
  "cny",
  "czk",
  "dkk",
  "eur",
  "gbp",
  "hkd",
  "huf",
  "idr",
  "ils",
  "inr",
  "jpy",
  "krw",
  "kwd",
  "lkr",
  "mmk",
  "mxn",
  "myr",
  "ngn",
  "nok",
  "nzd",
  "php",
  "pkr",
  "pln",
  "rub",
  "sar",
  "sek",
  "sgd",
  "thb",
  "try",
  "twd",
  "uah",
  "vef",
  "vnd",
  "zar",
  "xdr",
  "xag",
  "xau",
  "bits",
  "sats",
]
`;

exports[`CoinGeckoClient test Events /events/types 1`] = `
Object {
  "count": 3,
  "data": Array [
    "Event",
    "Conference",
    "Meetup",
  ],
}
`;

exports[`CoinGeckoClient test coins /coins/{id}/history should successful 1`] = `
Object {
  "community_data": Object {
    "facebook_likes": null,
    "reddit_accounts_active_48h": "9113.07692307692",
    "reddit_average_comments_48h": 1072.333,
    "reddit_average_posts_48h": 6.917,
    "reddit_subscribers": 2648989,
    "twitter_followers": 83714,
  },
  "developer_data": Object {
    "closed_issues": 5543,
    "code_additions_deletions_4_weeks": Object {
      "additions": 1942,
      "deletions": -1331,
    },
    "commit_count_4_weeks": 332,
    "forks": 28208,
    "pull_request_contributors": 720,
    "pull_requests_merged": 8450,
    "stars": 51244,
    "subscribers": 3752,
    "total_issues": 6109,
  },
  "id": "bitcoin",
  "image": Object {
    "small": "https://assets.coingecko.com/coins/images/1/small/bitcoin.png?1547033579",
    "thumb": "https://assets.coingecko.com/coins/images/1/thumb/bitcoin.png?1547033579",
  },
  "localization": Object {
    "ar": "بيتكوين",
    "de": "Bitcoin",
    "en": "Bitcoin",
    "es": "Bitcoin",
    "fr": "Bitcoin",
    "hu": "Bitcoin",
    "id": "Bitcoin",
    "it": "Bitcoin",
    "ja": "ビットコイン",
    "ko": "비트코인",
    "nl": "Bitcoin",
    "pl": "Bitcoin",
    "pt": "Bitcoin",
    "ro": "Bitcoin",
    "ru": "биткоин",
    "sv": "Bitcoin",
    "th": "บิตคอยน์",
    "tr": "Bitcoin",
    "vi": "Bitcoin",
    "zh": "比特币",
    "zh-tw": "比特幣",
  },
  "market_data": Object {
    "current_price": Object {
      "aed": 216047.9777669398,
      "ars": 5410316.137126201,
      "aud": 77443.71866877718,
      "bch": 108.88867356866139,
      "bdt": 4986355.506661704,
      "bhd": 22175.855659776604,
      "bits": 1000948.6777615082,
      "bmd": 58817.3738884188,
      "bnb": 194.4890148850541,
      "brl": 331306.5036386855,
      "btc": 1,
      "cad": 73902.559856451,
      "chf": 55518.5426565129,
      "clp": 42377917.88660569,
      "cny": 385412.60587864177,
      "czk": 1309551.1844134755,
      "dkk": 373053.9580945811,
      "dot": 1594.587856814039,
      "eos": 12269.41966053906,
      "eth": 30.733975464038206,
      "eur": 50151.45728919469,
      "gbp": 42667.71108775401,
      "hkd": 457287.6132224112,
      "huf": 18166929.56627026,
      "idr": 856839739.3317074,
      "ils": 196144.17844309928,
      "inr": 4305651.922062717,
      "jpy": 6514308.696599343,
      "krw": 66357761.22091409,
      "kwd": 17784.903429510647,
      "link": 2020.3781971342075,
      "lkr": 11731566.76407089,
      "ltc": 299.26808281595885,
      "mmk": 82914921.02210093,
      "mxn": 1202021.8496444104,
      "myr": 243886.24082832862,
      "ngn": 23259739.637827985,
      "nok": 502903.54517632176,
      "nzd": 84241.41902118336,
      "php": 2854453.6793477093,
      "pkr": 8977904.948130723,
      "pln": 232283.69038560346,
      "rub": 4453392.754385964,
      "sar": 220610.91199845573,
      "sats": 100094867.77615081,
      "sek": 513631.3636345784,
      "sgd": 79113.19100922598,
      "thb": 1837631.5064827378,
      "try": 485694.22856768314,
      "twd": 1669789.954267881,
      "uah": 1637263.2995164648,
      "usd": 58817.3738884188,
      "vef": 5889.383647447377,
      "vnd": 1358038191.0131512,
      "xag": 2404.7736228987765,
      "xau": 34.43227884801922,
      "xdr": 41476.48281439182,
      "xlm": 144402.42073839266,
      "xrp": 103349.20237563681,
      "yfi": 1.6291747927799918,
      "zar": 869847.1427497568,
    },
    "market_cap": Object {
      "aed": 4033641238570.1426,
      "ars": 101011240697450.36,
      "aud": 1445883365904.8728,
      "bch": 2032964384.3903894,
      "bdt": 93095845699323.8,
      "bhd": 414025841918.99695,
      "bits": 18687829925751.332,
      "bmd": 1098127310946.8978,
      "bnb": 3631132857.6077166,
      "brl": 6185531517101.6875,
      "btc": 18670118,
      "cad": 1379769513022.0032,
      "chf": 1036537742585.1293,
      "clp": 791200727537238.9,
      "cny": 7195698830441.736,
      "czk": 24449475140039.34,
      "dkk": 6964961417992.885,
      "dot": 29771143448.085213,
      "eos": 229071512853.78418,
      "eth": 573806948.5226982,
      "eur": 936333625461.225,
      "gbp": 796611200798.2758,
      "hkd": 8537613698800.777,
      "huf": 339178718699954.56,
      "idr": 15997299040412218,
      "ils": 3662034956545.72,
      "inr": 80387029451837.73,
      "jpy": 121622912053935.94,
      "krw": 1238907232210290.2,
      "kwd": 332046245647.56854,
      "link": 37720699345.122925,
      "lkr": 219029735810081.7,
      "ltc": 5587370419.807723,
      "mmk": 1548031359443305,
      "mxn": 22441889771439.395,
      "myr": 4553384894841.3125,
      "ngn": 434262083687525.6,
      "nok": 9389268531060.26,
      "nzd": 1572797233612.9377,
      "php": 53292987018955.89,
      "pkr": 167618544774384.47,
      "pln": 4336763908974.6826,
      "rub": 83145368224730.97,
      "sar": 4118831759098.7837,
      "sats": 1868782992575133,
      "sek": 9589558167558.488,
      "sgd": 1477052611498.788,
      "thb": 34308797066550.484,
      "try": 9067968559277.615,
      "twd": 31175175481395.94,
      "uah": 30567898999041.746,
      "usd": 1098127310946.8978,
      "vef": 109955487645.11293,
      "vnd": 25354733274722068,
      "xag": 44897407302.80765,
      "xau": 642854709.1014229,
      "xdr": 774370828369.6674,
      "xlm": 2696010234671.439,
      "xrp": 1929541803559.0193,
      "yfi": 30416885.623827998,
      "zar": 16240148797100.805,
    },
    "total_volume": Object {
      "aed": 211193735936.2901,
      "ars": 5288755254301.331,
      "aud": 75703686003.03023,
      "bch": 106442124.61419189,
      "bdt": 4874320320157.72,
      "bhd": 21677600747.66509,
      "bits": 978459011382.104,
      "bmd": 57495844477.9185,
      "bnb": 190119167.4029577,
      "brl": 323862592775.2194,
      "btc": 978459.0113821039,
      "cad": 72242091190.39264,
      "chf": 54271132544.52992,
      "clp": 41425755946340.23,
      "cny": 376753020110.4565,
      "czk": 1280127728547.5095,
      "dkk": 364672050764.6007,
      "dot": 1558760096.9007165,
      "eos": 11993746031.146273,
      "eth": 30043433.710943747,
      "eur": 49024636735.91986,
      "gbp": 41709037972.08301,
      "hkd": 447013114550.0063,
      "huf": 17758748613379.387,
      "idr": 837587963185421.1,
      "ils": 191737142164.9629,
      "inr": 4208911022729.512,
      "jpy": 6367942921276.467,
      "krw": 64866811739987.65,
      "kwd": 17385305974.01062,
      "link": 1974983630.3364513,
      "lkr": 11467977802425.322,
      "ltc": 292544022.44196784,
      "mmk": 81051959460643.12,
      "mxn": 1175014400631.427,
      "myr": 238406519127.68912,
      "ngn": 22737131639887.062,
      "nok": 491604131726.5445,
      "nzd": 82348653236.87666,
      "php": 2790318811726.366,
      "pkr": 8776186227125.959,
      "pln": 227064658862.59692,
      "rub": 4353332362152.287,
      "sar": 215654148559.19824,
      "sats": 97845901138210.39,
      "sek": 502090913792.5612,
      "sgd": 77335648052.6914,
      "thb": 1796342956502.83,
      "try": 474781480086.5948,
      "twd": 1632272527221.4238,
      "uah": 1600476692770.838,
      "usd": 57495844477.9185,
      "vef": 5757058907.573982,
      "vnd": 1327525312056481.5,
      "xag": 2350742324.6928253,
      "xau": 33658642.31581824,
      "xdr": 40544574633.87168,
      "xlm": 141157936441.70767,
      "xrp": 101027116205.14148,
      "yfi": 1592569.9214440342,
      "zar": 850303111695.8679,
    },
  },
  "name": "Bitcoin",
  "public_interest_stats": Object {
    "alexa_rank": 5018,
    "bing_matches": null,
  },
  "symbol": "btc",
}
`;

exports[`CoinGeckoClient test coins /coins/market should successful 1`] = `
Array [
  Object {
    "ath": 3.35,
    "ath_change_percentage": -43.66066,
    "ath_date": "2021-04-08T08:34:23.319Z",
    "atl": 0.103356,
    "atl_change_percentage": 1727.29733,
    "atl_date": "2020-01-30T16:40:02.103Z",
    "circulating_supply": 211555966.02,
    "current_price": 1.89,
    "fully_diluted_valuation": null,
    "high_24h": 2.07,
    "id": "origin-protocol",
    "image": "https://assets.coingecko.com/coins/images/3296/large/op.jpg?1547037878",
    "last_updated": "2021-04-19T11:31:46.327Z",
    "low_24h": 1.76,
    "market_cap": 400619666,
    "market_cap_change_24h": 3454096,
    "market_cap_change_percentage_24h": 0.86969,
    "market_cap_rank": 160,
    "max_supply": null,
    "name": "Origin Protocol",
    "price_change_24h": 0.04292302,
    "price_change_percentage_24h": 2.31819,
    "roi": Object {
      "currency": "usd",
      "percentage": 1288.925192137712,
      "times": 12.889251921377118,
    },
    "symbol": "ogn",
    "total_supply": 1000000000,
    "total_volume": 155393589,
  },
  Object {
    "ath": 0.00012728,
    "ath_change_percentage": -52.4766,
    "ath_date": "2021-04-14T09:27:42.881Z",
    "atl": 0.00000177,
    "atl_change_percentage": 3313.97663,
    "atl_date": "2020-10-23T19:10:35.677Z",
    "circulating_supply": 0,
    "current_price": 0.00006049,
    "fully_diluted_valuation": null,
    "high_24h": 0.00011275,
    "id": "bitcorn",
    "image": "https://assets.coingecko.com/coins/images/10678/large/BITCORN-logo.png?1581803042",
    "last_updated": "2021-04-19T07:22:53.760Z",
    "low_24h": 0.00005457,
    "market_cap": 0,
    "market_cap_change_24h": 0,
    "market_cap_change_percentage_24h": 0,
    "market_cap_rank": null,
    "max_supply": null,
    "name": "BITCORN",
    "price_change_24h": -0.00005226,
    "price_change_percentage_24h": -46.35164,
    "roi": null,
    "symbol": "corn",
    "total_supply": 100000000000,
    "total_volume": 376.91,
  },
]
`;

exports[`CoinGeckoClient test simple /simple/supported_vs_currencies should successful 1`] = `
Array [
  "btc",
  "eth",
  "ltc",
  "bch",
  "bnb",
  "eos",
  "xrp",
  "xlm",
  "link",
  "dot",
  "yfi",
  "usd",
  "aed",
  "ars",
  "aud",
  "bdt",
  "bhd",
  "bmd",
  "brl",
  "cad",
  "chf",
  "clp",
  "cny",
  "czk",
  "dkk",
  "eur",
  "gbp",
  "hkd",
  "huf",
  "idr",
  "ils",
  "inr",
  "jpy",
  "krw",
  "kwd",
  "lkr",
  "mmk",
  "mxn",
  "myr",
  "ngn",
  "nok",
  "nzd",
  "php",
  "pkr",
  "pln",
  "rub",
  "sar",
  "sek",
  "sgd",
  "thb",
  "try",
  "twd",
  "uah",
  "vef",
  "vnd",
  "zar",
  "xdr",
  "xag",
  "xau",
  "bits",
  "sats",
]
`;
